IoTuring/Configurator/MenuPreset.py

```diff
@@ -97,42 +126,74 @@ def AddEntry(self, name, key, default=None, mandatory=False, display_if_key_valu
                 default=default,
                 mandatory=mandatory,
                 dependsOn=display_if_key_value,
-                modify_value_callback=modify_value_callback
+                instruction=instruction,
+                question_type=question_type,
+                choices=choices
             ))
 
     def AskQuestions(self) -> None:
         """Ask all questions of this preset"""
         for q_preset in self.presets:
+            # if the previous question was cancelled:
+
+            if self.cancelled:
+                raise UserCancelledException
+
             try:
-                value = None
 
                 # It should be displayed, ask question:
                 if q_preset.ShouldDisplay(self):
-                    value = input(q_preset.question)
 
-                    # Mandatory loop:
-                    while value == "" and q_preset.mandatory:
-                        value = input(
-                            "You must provide a value for this key: ")
+                    question_options = {}
 
-                    # Set default:
-                    if value == "":
-                        value = q_preset.default
+                    if q_preset.mandatory:
+                        def validate(x): return bool(x)
+                        question_options.update({
+                            "validate": validate,
+                            "invalid_message": "You must provide a value for this key"
+                        })
 
-                # It should not be displayed:
-                else:
-                    # It's already answered:
-                    if self.GetAnsweredPresetByKey(q_preset.key):
-                        continue
+                    # text:
+                    prompt_function = inquirer.text
 
-                    # Set default value otherwise:
-                    else:
-                        value = q_preset.default
+                    question_options["message"] = q_preset.question + ":"
 
-                # Set and sanitize the value:
-                q_preset.SetValue(value)
-                # Add to answered questions:
-                self.results.append(q_preset)
+                    if q_preset.default is not None:
+                        question_options["default"] = q_preset.default
```

Comment:
default value for yes-no questions should be converted to boolean here, not in `IoTuring/Configurator/ConfiguratorObject.py`